rust/ddlog_benches/README.md

```diff
@@ -0,0 +1,25 @@
+# ddlog_benches
+
+DDlog has a unique benchmarking scenario since it's a compiler, so benchmarking the programs
+it generates is a little more involved than normal, which is why we use [`cargo-make`] to automate
+the setup, build and bench process in a cross-platform way. To run the benchmarks in this crate,
+first install `cargo-make` with the following command (`--force` makes cargo install the latest version)
+
+```sh
+cargo install --force cargo-make
+```
+
+To run all benchmarks use the following command
+
+```sh
+cargo make benchmark-all
+```
+
+When adding a benchmark, add the required ddlog code to `/ddlog`, any datasets to `/data` (preferably in a
```

Comment:
What I was actually considering was just downloading the sources if they weren't already there (and providing a command to re-download them as well) since they're available as part of the snap dataset

Reply:
Sounds good, there's no reason to create a new repo if we can use an external source (as long as the external source does not change and is likely to stay there for a long time). Do you plan to integrate the download step in Makefile.toml?

Reply:
Yes, the [snap dataset](http://snap.stanford.edu/data/index.html) shouldn't change and the downloading will be in the makefile

Reply:
Considering the task a wee bit more, I may opt for an [xtask](https://github.com/matklad/cargo-xtask/) rig to minimize code duplication, as much as I like cargo-make it's not as modular as I'd like

Reply:
I'm not familiar with either of those, but xtask does not seem to have many users or be particularly well maintained.

Reply:
Ah, it's a pattern as opposed to an actual crate, it's just a pattern of using rust as a medium for setup/teardown scripts. I don't actually think it's needed though